src/commands/synthetics/api.ts

```diff
@@ -259,27 +265,43 @@ const createMobileVersion = (request: (args: AxiosRequestConfig) => AxiosPromise
   return resp.data
 }
 
-type RetryPolicy = (retries: number, error: AxiosError) => number | undefined
+const retryWithJitter = (delay: number = DELAY_FOR_TOO_MANY_REQUESTS) => delay + Math.floor(Math.random() * delay)
+
+export type OptionalRetries = {
+  retryOn404?: boolean | undefined
+  retryOn429?: boolean | undefined
+}
 
-const retryOn5xxErrors: RetryPolicy = (retries, error) => {
-  // Retry on Node.js errors for both retry policies.
+export const determineRetryDelay = (
+  retries: number,
+  error: Error,
+  optionalRetries: OptionalRetries = {retryOn404: false, retryOn429: false}
+) => {
+  // Retry on 5xx
   if (retries < MAX_RETRIES && isNodeError(error)) {
     return LARGE_DELAY_BETWEEN_RETRIES
   }
 
-  if (retries < MAX_RETRIES && is5xxError(error)) {
+  if (retries < MAX_RETRIES && is5xxError(error as AxiosError<unknown, any> | EndpointError)) {
```

Comment:
I'm not sure if there's not a clear way to do this 🤔

Reply:
Yes there is. Maybe `is5xxError()` should take either a `Error` or `unknown` and just return a boolean?

Reply:
Tried it, but in the end we can't cuz later one we try to do `error.response` when trying to get the http status and `Property 'response' does not exist on type 'Error'`

Reply:
Also i was using the 5xx as an example but we do the same casting for 404 and 429